src/seedu/addressbook/AddressBook.java

```diff
@@ -303,16 +302,43 @@ private static void setupDefaultFileForStorage() {
         storageFilePath = DEFAULT_STORAGE_FILEPATH;
         createFileIfMissing(storageFilePath);
     }
-
+    
     /**
-     * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
-     * TODO: Implement a more rigorous validity checking.
+     * Returns true if the given file path is acceptable.
+     * The file path is acceptable if it has a folder that exists,
+     * has a file name that is acceptable to the operating system,
+     * File name must also contain an extentsion
      */
     private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+        if (filePath == null) {
+            return false;
+        }
+        File filePathToValidate = new File(filePath);
+        return hasValidFileName(filePathToValidate) && hasValidFolder(filePathToValidate);
     }
-
+    
+    /**
+     * Returns true if the file path has a folder that exists
+     */
+    private static boolean hasValidFolder(File filePath) {
+        File folderToValidate = filePath.getAbsoluteFile().getParentFile();
+        return folderToValidate.exists();
+    }
+    
+    /**
+     * Returns true if file path has a valid file name.
+     * File name is valid if it has an extension and no reserved characters.
+     */
+    private static boolean hasValidFileName(File filePath) {
+        try {
+            File fileNameToValidate = new File(filePath.getName()).getCanonicalFile();
```

Comment:
Ok got it. Thanks for the explanation, I've learnt something! =)
Will change it to Paths.get().